the-collab-lab/tcl-22-smart-shopping-list
# Estimate the next purchase date and add it to the database

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



This feature retrieves the data about previous purchases of an item from the database; uses it to estimate the number of days until the next purchase; saves the estimate to the database.

## Related Issue


Closes #11  

## Acceptance Criteria



* When a purchase is recorded, the estimated number of days until the next purchase date should be calculated and recorded in the database
* The script at /src/lib/estimates.js should be used to calculate the next estimated purchase interval.

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

## Testing Steps / QA Criteria


To test the feature, use one of the following ways:

1) Follow the link to see the preview: https://tcl-22-shopping-list--pr38-ab-dt-estimate-43xqv8qs.web.app
2) Launch the app locally from your terminal:
* Pull down this branch with `git pull origin ab-dt-estimate` and check that branch out with `git checkout origin ab-dt-estimate`
* Run `npm install` to update the dependencies, then `npm start`  to launch the app in your browser at [http://localhost:3000](http://localhost:3000)

When on application:

* For testing purposes, we created a list with items that have a few purchase dates in the past. To join this list, please use token `tour dobson oath`. 
* Open the list in [Firebase](https://console.firebase.google.com/u/2/project/tcl-22-shopping-list/firestore/data~2Ftour%20dobson%20oath)
* In the app, mark an item as purchased
* In Firebase, the current date should be added to `purchaseDates` and a field `nextEstimate` should update (or appear, if it's the first time this item is marked as purchased).
* `nextEstimate` should be an integer. The value should decrease each time the item is marked as purchased.
